### DISCUSSION TITLE
Copilot importing indefinitely.

### DISCUSSION THREAD

**Discussion by kenan238** [community/community#7516] [Copilot Conversations] [answered]
https://user-images.githubusercontent.com/73741117/140978729-59030a58-e5da-4639-ac15-ae086a1d93b6.mp4

When using an _import statement_ in python, github copilot keeps recommending to import *random modules* and sometimes repeating imports, because of _file size limitations_ i could only show a _few lines_ of import statements, but i tried it 15 mins ago and it kept going, at the end i got tired and stopped, it got to hundreds of lines of just importing random stuff. Please fix.

**Comment by Krzysztof-Cieslak** [comment, marked as answer]
Copilot doesn't (yet) read in your mind. 😄  However, it's really good at recognizing patterns - if it seems file only with `import` statements, it will definitely try to come up with yet another `import` statement. 

Essentially, if you don't provide it with enough context it won't know what to do.

**Comment by kenan238** [comment]
Thanks, i understand, but i think it can _not repeat_ imports?
Also, copilot can just stop importing to a point because it doesn't know what the user is going to write, so my suggestion here is to:

1. Make copilot not import definitely and just stop at a point to let the user write.
2. Not repeat imports because its stupid.

Btw when you said: Copilot doesn't **(yet)** read in your mind.
that yet made me start to be scared a bit 

Anyways thanks for the reply!

**Comment by PetrDlouhy** [comment]
Well in many cases I first use some function/class and then go up and try to import it. That is case where the Copilot could have guessed what I want easily.
Either based on what I written recently or by seeing the whole file and knowing what imports are missing.
